File: Py/solved/s0001.py
# https://leetcode.com/problems/two-sum/description/
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap: dict[int, int] = {}
        for idx, val in enumerate(nums):
            matchedIdx = numMap.get(target - val)
            if matchedIdx is not None:
                return [matchedIdx, idx]
            else:
                numMap[val] = idx
        return []

File: csharp/LeetCode/Solutions/S0001.cs
namespace LeetCode.Solutions;
public class S0001
{
    public int[] TwoSum(int[] nums, int target)
    {
        var dict = new Dictionary<int, int>(); // val, idx
        for (int i = 0; i < nums.Length; i++)
        {
            var n = nums[i];
            var isMatch = dict.TryGetValue(target - n, out var idx);
            if (isMatch) return new int[] { idx, i };
            else dict[n] = i;
        }
        return Array.Empty<int>();
    }
}
